### PR TITLE
Added in round trip token tests for JsonWebKey and RsaSecurityKey

### LINKED ISSUE #797: Add in RoundTripToken tests using JsonWebKeys and RsaSecurityKeys for signing and validation
Tests need to be added to both SamlSecurityTokenHandlerTests.cs and Saml2SecurityTokenHandlerTests.cs.

### PR DESCRIPTION
**Pull request by mafurman**
Fixes #797.

### CODE REVIEW

File: test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs [hunk 1]
```diff
@@ -317,6 +317,63 @@ public static TheoryData<Saml2TheoryData> RoundTripTokenTheoryData
                         ValidAudience = Default.Audience,
                         ValidIssuer = Default.Issuer,
                     }
+                },
+                    new Saml2TheoryData
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs]: tabbing.

File: test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs [hunk 2]
```diff
@@ -317,6 +317,63 @@ public static TheoryData<Saml2TheoryData> RoundTripTokenTheoryData
...
+                    ValidationParameters = new TokenValidationParameters
+                    {
+                        IssuerSigningKey = KeyingMaterial.RSASigningCreds_2048_Public.Key,
+                        ValidAudience = Default.Audience,
+                        ValidIssuer = Default.Issuer,
+                    },
+                },
+                new Saml2TheoryData
+                {
+                    First = true,
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs]: only one should be marked first

File: test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs [hunk 3]
```diff
@@ -317,6 +317,63 @@ public static TheoryData<Saml2TheoryData> RoundTripTokenTheoryData
...
+                    TestId = nameof(Default.ClaimsIdentity) + nameof(KeyingMaterial.JsonWebKeyRsa256SigningCredentials),
+                    TokenDescriptor = new SecurityTokenDescriptor
+                    {
+                        Expires = DateTime.UtcNow + TimeSpan.FromDays(1),
+                        Audience = Default.Audience,
+                        SigningCredentials = KeyingMaterial.JsonWebKeyRsa256SigningCredentials,
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs]: If you can keep things in order, it is easier to find. It's natural to get out of order but a, b, c is easier to see follow.

File: test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs (hunk 2 above)
Comment (mafurman) [reply test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs]: Got it!

File: test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs (hunk 1 above)
Comment (mafurman) [reply test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs]: Fixed.

**Pull request merged**